framework/test/ResourceCompilerTest.cpp

```diff
@@ -0,0 +1,448 @@
+  testing::MakeDirectory(rc1dir_path);
+  testing::MakeDirectory(rc2dir_path);
+  std::string rc1file_path(rc1dir_path + testing::DIR_SEP + "resource1.txt");
+  std::string rc2file_path(rc2dir_path + testing::DIR_SEP + "resource2.txt");
+  std::ofstream rc1file(rc1file_path.c_str());
+  std::ofstream rc2file(rc2file_path.c_str());
+  if (!rc1file.is_open())
+  {
+    throw std::runtime_error("Couldn't open " + rc1file_path);
+  }
+  if (!rc2file.is_open())
+  {
+    throw std::runtime_error("Couldn't open " + rc2file_path);
+  }
+  rc1file << resource1_txt << std::endl;
+  rc1file.close();
+  rc2file << resource2_txt << std::endl;
+  rc2file.close();
+
+  // Create 2 binary files filled with random numbers
+  // to test bundle-file functionality.
+  auto create_mock_dll = [](const std::string& tempdir,
+                            const std::string& dllname,
+                            const std::array<char, 5>& dat)
+  {
+    std::string dll_path(tempdir + testing::DIR_SEP + dllname);
+    std::ofstream dll(dll_path.c_str());
+    if (!dll.is_open())
+    {
+      throw std::runtime_error("Couldn't open " + dll_path);
+    }
+    dll.write(dat.data(), dat.size()); //binary write
+    dll.close();
+  };
+  std::array<char, 5> data1 = { {2, 4, 6, 8, 10} };
+  std::array<char, 5> data2 = { {1, 2, 3, 4, 5} };
+  create_mock_dll(tempdir, "sample.dll", data1);
+  create_mock_dll(tempdir, "sample1.dll", data2);
+}
+
+/*
+* Use resource compiler to create Example.zip with just manifest.json
+*/
+static void testManifestAdd(const std::string& rcbinpath, const std::string& tempdir)
+{
+  std::ostringstream cmd;
+  cmd << rcbinpath;
+  cmd << " --bundle-name " << "mybundle";
+  cmd << " --out-file " << tempdir << testing::DIR_SEP << "Example.zip";
+  cmd << " --manifest-add " << tempdir << testing::DIR_SEP << "manifest.json";
+  std::system(cmd.str().c_str());
+
+  ZipFile zip(tempdir + testing::DIR_SEP + "Example.zip");
+  US_TEST_CONDITION(zip.size() == 2, "Check number of entries of zip.")
+  US_TEST_CONDITION(zip[0].name == "mybundle/manifest.json", "Check name of entry 0.")
+  US_TEST_CONDITION(zip[1].name == "mybundle/", "Check name of entry 1.")
+}
+
+/*
+* Use resource compiler to create Example.zip with manifest.json and
+* one --res-add option
+*/
+static void testManifestResAdd(const std::string& rcbinpath, const std::string& tempdir)
+{
+  std::ostringstream cmd;
+  cmd << rcbinpath;
+  cmd << " --bundle-name mybundle ";
+  cmd << " --out-file Example2.zip ";
+  cmd << " --manifest-add manifest.json ";
+  cmd << " --res-add resource1/resource1.txt";
+
+  auto cwdir = testing::GetCurrentWorkingDirectory();
+  testing::ChangeDirectory(tempdir);
+  std::system(cmd.str().c_str());
+  testing::ChangeDirectory(cwdir);
+
+  ZipFile zip(tempdir + testing::DIR_SEP + "Example2.zip");
+  US_TEST_CONDITION(zip.size() == 4, "Check number of entries of zip.")
+  US_TEST_CONDITION(zip[0].name == "mybundle/manifest.json", "Check name of entry 0.")
```

Comment:
The test assumes the zip file contents are retrieved in a particular order. If the order of insertion changes in the implementation, the test needs to change. It is preferable to check for existence irrespective of the order.